packages/consumption/src/modules/requests/utility/validateRelationshipAttributesWithinRequest.ts

```diff
@@ -0,0 +1,206 @@
+    }
+
+    return ValidationResult.success();
+}
+
+function extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(requestItemGroup: RequestItemGroup, recipient?: CoreAddress): RelationshipAttributeFragment[] | void {
+    const fragmentsOfMustBeAcceptedItemsOfGroup: RelationshipAttributeFragment[] = [];
+
+    for (const item of requestItemGroup.items) {
+        if (item instanceof RequestItemGroup) {
+            const fragments = extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(item, recipient);
+            if (fragments) fragmentsOfMustBeAcceptedItemsOfGroup.push(...fragments);
+        } else {
+            const fragment = extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(item, recipient);
+            if (fragment) fragmentsOfMustBeAcceptedItemsOfGroup.push(fragment);
+        }
+    }
+
+    if (fragmentsOfMustBeAcceptedItemsOfGroup.length !== 0) return fragmentsOfMustBeAcceptedItemsOfGroup;
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(requestItem: RequestItem, recipient?: CoreAddress): RelationshipAttributeFragment | void {
+    if (requestItem.mustBeAccepted) {
+        return extractRelationshipAttributeFragmentFromRequestItem(requestItem, recipient);
+    }
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentsFromAcceptedItemsOfGroup(
+    requestItemGroup: RequestItemGroup,
+    decideGroupParams: DecideRequestItemGroupParametersJSON
+): RelationshipAttributeFragment[] | void {
+    const fragmentsOfAcceptedItemsOfGroup: RelationshipAttributeFragment[] = [];
+
+    for (let i = 0; i < decideGroupParams.items.length; i++) {
+        const decideItemParams = decideGroupParams.items[i];
+        const item = requestItemGroup.items[i];
+
+        if (item instanceof RequestItem) {
+            const fragment = extractRelationshipAttributeFragmentFromAcceptedRequestItem(item, decideItemParams);
+            if (fragment) fragmentsOfAcceptedItemsOfGroup.push(fragment);
+        }
+    }
+
+    if (fragmentsOfAcceptedItemsOfGroup.length !== 0) return fragmentsOfAcceptedItemsOfGroup;
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentFromAcceptedRequestItem(
+    requestItem: RequestItem,
+    decideItemParams: DecideRequestItemParametersJSON
+): RelationshipAttributeFragment | void {
+    if (decideItemParams.accept) {
+        return extractRelationshipAttributeFragmentFromRequestItem(requestItem);
+    }
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentFromRequestItem(requestItem: RequestItem, recipient?: CoreAddress): RelationshipAttributeFragment | void {
+    if (requestItem instanceof CreateAttributeRequestItem && requestItem.attribute instanceof RelationshipAttribute) {
+        const ownerIsEmptyString = requestItem.attribute.owner.toString() === "";
+        return {
+            owner: ownerIsEmptyString && recipient ? recipient.toString() : requestItem.attribute.owner.toString(),
```

Comment:
I think there's a scenario where the recipient remains unnamed and is named as attributeOwner in one request item but not named in the other request item. In both cases the recipient would be the owner but the strings would be different. What about having the owner be a boolean - request sender vs. recipient?

Reply:
Can you explain why this scenario should not be covered? Either the recipient is clearly named as the owner, or the empty string is used as a placeholder. If this is the case, the address of the recipient is used instead, so that you still have comparability.

Reply:
Ah, I wasn't aware of how the validation for the attributeOwner worked - I thought you could put the address of the recipient as the owner without giving that address as recipient. So it works the way it is.

In my scenario I meant that the recipient is undefined, this might have been unclear.